### PR TITLE
refactor(mini-rx-store): store can be have a custom combineReducers fn

### PR DESCRIPTION
**Pull request by BrainCrumbz**
See also:
* https://github.com/spierala/mini-rx-store/discussions/148
* https://github.com/spierala/mini-rx-store/pull/155

### CODE REVIEW

File: libs/mini-rx-store/src/lib/spec/store.spec.ts [hunk 1]
```diff
@@ -1,10 +1,17 @@
...
+    AppState,
+    Reducer,
+    ReducerDictionary,
+    StoreExtension,
+} from '../models';
 import { createFeatureSelector, createSelector } from '../selector';
 import { Observable, of } from 'rxjs';
 import { ofType } from '../utils';
-import { catchError, map, mapTo, mergeMap, take, withLatestFrom } from 'rxjs/operators';
+import { catchError, map, mapTo, mergeMap, reduce, take, withLatestFrom } from 'rxjs/operators';
```
Comment (spierala) [inline libs/mini-rx-store/src/lib/spec/store.spec.ts:14]: reduce not used.

But nice operator, I did not know that it exists :)

File: libs/mini-rx-store/src/lib/spec/store.spec.ts [hunk 2]
```diff
@@ -386,6 +394,27 @@ describe('Store Config', () => {
             );
         });
     });
+
+    it('should call custom combineReducer fn', () => {
```
Comment (spierala) [inline libs/mini-rx-store/src/lib/spec/store.spec.ts:398]: Great, also a nice test!

File: libs/mini-rx-store/src/lib/store-core.ts [hunk 3]
```diff
@@ -54,11 +56,13 @@ class StoreCore {
     private extensions: StoreExtension[] = [];
 
     constructor() {
+        this.combineReducersFn = defaultCombineReducers;
```
Comment (spierala) [inline libs/mini-rx-store/src/lib/store-core.ts:59]: Super minor, but I would rather initialise the `private combineReducersFn` at declaration on line 37.

`private combineReducersFn: CombineReducersFn<AppState> = defaultCombineReducers;`

A little bit more consistent with the other props.

**Review by spierala — COMMENTED** [review]
Great PR! I looks already very nice. Just minor remarks.

Could you maybe adjust the commit message?
It is rather a new feature, not just a refactor.

refactor(mini-rx-store) should be feat(mini-rx-store)

Then the automatic CHANGELOG creation will list your new feature. Refactors are ignored. 
With _GIT amend commit_ you can easily overwrite the last commit message. Then force push :)

**Comment by BrainCrumbz** [comment]
yes, it's a feature. *refactor* was probably a copy-paste typo

File: libs/mini-rx-store/src/lib/store-core.ts (hunk 3 above)
Comment (BrainCrumbz) [reply libs/mini-rx-store/src/lib/store-core.ts:59]: yes, we generally tried to be consistent with the style.

**Comment by BrainCrumbz** [comment]
ok, branch has been force-pushed with mentioned changes

**Comment by spierala** [comment]
@all-contributors please add @BrainCrumbz for code!

**Comment by spierala** [comment]
@GiuseppePiscopo mini-rx-store@4.2 is published: https://www.npmjs.com/package/mini-rx-store

Thank you for the contribution!

**Comment by BrainCrumbz** [comment]
You're welcome!

**Pull request merged**